File: app/src/main/java/mariavv/fitnesspal/presentation/handbook/HandBookFragment.java
package mariavv.fitnesspal.presentation.handbook;

import android.database.Cursor;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.widget.DividerItemDecoration;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.arellomobile.mvp.MvpAppCompatFragment;
import com.arellomobile.mvp.presenter.InjectPresenter;

import mariavv.fitnesspal.R;

public class HandBookFragment extends MvpAppCompatFragment implements HandBookView {

    private View view;
    private RecyclerView recycler;

    @InjectPresenter
    HandBookPresenter presenter;
    private FeedAdapter adapter;

    public static HandBookFragment newInstance() {
        return new HandBookFragment();
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        view = inflater.inflate(R.layout.fragment_hand_book, container, false);

        configureViews();

        return view;
    }

    @Override
    public void updateFeed(Cursor c) {
        adapter.updateItems(c);
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
    }

    private void configureViews() {
        recycler = view.findViewById(R.id.recycler);
        configureRecyclerView();
    }

    private void configureRecyclerView() {
        adapter = new FeedAdapter();
        recycler.setAdapter(adapter);
        recycler.setLayoutManager(new LinearLayoutManager(getContext()));

        DividerItemDecoration decoration = new DividerItemDecoration(recycler.getContext(), DividerItemDecoration.VERTICAL);
        recycler.addItemDecoration(decoration);
    }
}

File: app/src/main/java/mariavv/fitnesspal/presentation/handbook/HandbookFragment.kt
package mariavv.fitnesspal.presentation.handbook

import android.database.Cursor
import android.os.Bundle
import android.support.v7.widget.DividerItemDecoration
import android.support.v7.widget.LinearLayoutManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.arellomobile.mvp.presenter.InjectPresenter
import kotlinx.android.synthetic.main.fragment_hand_book.*
import mariavv.fitnesspal.R
import mariavv.fitnesspal.other.Const
import mariavv.fitnesspal.other.FrmFabric
import mariavv.fitnesspal.presentation.common.BaseFragment

class HandbookFragment : BaseFragment(), HandBookView, FrmFabric.IFragment {

    private lateinit var adapter: FeedAdapter

    @InjectPresenter
    lateinit var presenter: HandBookPresenter

    override val name: String
        get() = Const.Screen.HANDBOOK

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_hand_book, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        adapter = FeedAdapter()
        recycler.adapter = adapter
        recycler.layoutManager = LinearLayoutManager(context)
        recycler.addItemDecoration(DividerItemDecoration(recycler.context, DividerItemDecoration.VERTICAL))

        fab.setOnClickListener { presenter.onFabClick() }
    }

    override fun onStart() {
        super.onStart()
        presenter.onStart()
    }

    override fun updateFeed(c: Cursor) {
        adapter.updateItems(c)
    }

    override fun onBackPressed() {
        presenter.onBackPressed()
    }
}
